drivers/modem/Kconfig

```diff
@@ -139,6 +139,24 @@ config MODEM_CELL_INFO
 	help
 	  Query for numerical operator id, location area code and cell id.
 
+config MODEM_SMS_IN_MSG_MAX_LEN
+        int "Maximum length of inbound SMS messages"
+        default 160
+        help
+          This is the maximum length of inbound (received) SMS messages.
+          Inbound SMS messages longer than this will be truncated.
+          The absolute maximum value may be network- and/or modem-dependent,
+          but a smaller value may be preferred in order to conserve memory.
+
+config MODEM_SMS_OUT_MSG_MAX_LEN
+        int "Maximum length of outbound SMS messages"
+        default 160
```

Comment:
I suppose the typical scenario is that both the inbound and outbound values are the same. So in order to avoid possible mismatch we could have here `default MODEM_SMS_IN_MSG_MAX_LEN`. Then user could override it if needed, what do you think?